release: bug: image signing still broken

We're making progress on this, but here's the latest issue:

```
docker trust sign brigadecore/brigade-acr-gateway:v0.4.0-rc.3
Error: No such image: brigadecore/brigade-acr-gateway:v0.4.0-rc.3
make: *** [Makefile:172: sign] Error 1
```

Again, this is because of buildx. With the build/push happening in another container, the image we're trying to sign isn't present locally. I had not presumed this to be a problem, but it is. (i.e. I had thought `docker trust sign ...` just hit up notary directly with no need to have the image present, but since there's a digest involved... I should have known better.)

We can fix this by pulling the image before we sign.

--- a/Makefile
+++ b/Makefile
@@ -169,8 +169,12 @@ push:
 
 .PHONY: sign
 sign:
+	docker pull $(DOCKER_IMAGE_NAME):$(IMMUTABLE_DOCKER_TAG)
+	docker pull $(DOCKER_IMAGE_NAME):$(MUTABLE_DOCKER_TAG)
 	docker trust sign $(DOCKER_IMAGE_NAME):$(IMMUTABLE_DOCKER_TAG)
 	docker trust sign $(DOCKER_IMAGE_NAME):$(MUTABLE_DOCKER_TAG)
+	docker trust inspect --pretty $(DOCKER_IMAGE_NAME):$(IMMUTABLE_DOCKER_TAG)
+	docker trust inspect --pretty $(DOCKER_IMAGE_NAME):$(MUTABLE_DOCKER_TAG)
 
 .PHONY: publish-chart
 publish-chart:

